src/ResourceManager/Automation/Commands.Automation.Test/ScenarioTests/ModuleTests.ps1

```diff
@@ -0,0 +1,175 @@
+$testAutomationAccount = @{
+    ResourceGroupName = 'anatolib-azureps-test-rg'
+    AutomationAccountName = 'anatolib-azureps-test-aa'
+}
+
+$testGlobalModule = @{
+	Name = 'Azure'
+	Version = '1.0.3'
+	Size = 41338511
+	ActivityCount = 673
+}
+
+$testNonGlobalModule = @{
+    Name = 'Pester'
+	Version = '3.0.3'
+    ContentLinkUri = 'https://devopsgallerystorage.blob.core.windows.net/packages/pester.3.0.3.nupkg'
+	Size = 74921
+}
+
+function EnsureTestModuleImported {
+	$foundModule = Get-AzureRmAutomationModule -Name $testNonGlobalModule.Name @testAutomationAccount -ErrorAction Ignore
+    if ($foundModule) {
+		if ($foundModule.ProvisioningState -ne 'Succeeded') {
+			Remove-AzureRmAutomationModule -Name $testNonGlobalModule.Name @testAutomationAccount -Force
+			$foundModule = $null
+		}
+	}
+
+    if (-not $foundModule) {
+        $output = New-AzureRmAutomationModule -Name $testNonGlobalModule.Name -ContentLinkUri $testNonGlobalModule.ContentLinkUri @testAutomationAccount
+		Write-Verbose "Module $($testNonGlobalModule.Name) provisioning state: $($output.ProvisioningState)"
+
+		$startTime = Get-Date
+		$timeout = New-TimeSpan -Minutes 3
+		$endTime = $startTime + $timeout
+
+        while ($output.ProvisioningState -ne 'Succeeded') {
+            [Microsoft.WindowsAzure.Commands.Utilities.Common.TestMockSupport]::Delay(10*1000)
+
+            $output = Get-AzureRmAutomationModule -Name $testNonGlobalModule.Name @testAutomationAccount
+			Write-Verbose "Module $($testNonGlobalModule.Name) provisioning state: $($output.ProvisioningState)"
+
+			if ((Get-Date) -gt $endTime) {
+				throw "Module $($testNonGlobalModule.Name) took longer than $timeout to import"
+			}
+        }
+    }
+}
+
+function Remove-TestNonGlobalModule {
+    if (Get-AzureRmAutomationModule -Name $testNonGlobalModule.Name @testAutomationAccount -ErrorAction Ignore) {
+        Remove-AzureRmAutomationModule -Name $testNonGlobalModule.Name @testAutomationAccount -Force
+    }
+}
+
+<#
+.SYNOPSIS
+Tests getting all modules from an Automation account.
+#>
+function Test-GetAllModules {
+	$output = Get-AzureRmAutomationModule @testAutomationAccount
```

Comment:
It is better to have the test create the account and resource group, I am sure they will bounce back because of that. I do noticed creating resource group through test seems to have some permission problem, but the code reviewer insist I increase permission on the test-cred to include adding resource group creation

Reply:
Do you know if we also need to remove them after each test?

Reply:
Ok, I found that other tests do
```
    finally
    {
        Clean-ResourceGroup $rgname
    }
```
So this is what I'm going to do as well